src/commands/createFunction/dotnetSteps/DotnetFunctionCreateStep.ts

```diff
@@ -66,22 +66,29 @@ export class DotnetFunctionCreateStep extends FunctionCreateStepBase<IDotnetFunc
             return;
         }
 
+        const packages: string[] = [];
+
+        switch (context.newDurableStorageType) {
+            case DurableBackend.Netherite:
+                packages.push(durableUtils.dotnetDfNetheritePackage);
+                break;
+            case DurableBackend.SQL:
+                packages.push(durableUtils.dotnetDfSqlPackage);
+                break;
+            case DurableBackend.Storage:
+            default:
+        }
+
+        // Seems that the package arrives out-dated and needs to be updated
+        packages.push(durableUtils.dotnetDfBasePackage);
+
         try {
-            switch (context.newDurableStorageType) {
-                case DurableBackend.Netherite:
-                    await cpUtils.executeCommand(ext.outputChannel, context.projectPath, 'dotnet', 'add', 'package', durableUtils.dotnetDfNetheritePackage);
-                    break;
-                case DurableBackend.SQL:
-                    await cpUtils.executeCommand(ext.outputChannel, context.projectPath, 'dotnet', 'add', 'package', durableUtils.dotnetDfSqlPackage);
-                    break;
-                case DurableBackend.Storage:
-                default:
+            while (packages.length) {
+                await cpUtils.executeCommand(ext.outputChannel, context.projectPath, 'dotnet', 'add', 'package', packages[packages.length - 1]);
+                packages.pop();  // Only pop off the package if we know running it did not cause an error
```

Comment:
This could result in an infinite loop

Reply:
Hmm, I'm not immediately seeing how.  If each pass is popping off a value and reducing the length by 1, and nothing is pushing values back into the array, it should eventually hit 0 right?  And 0 evaluates as false-y.  

Or am I missing something? 👀 😅

Reply:
🤦 I'm not thinking clearly. My brain was thinking that an error thrown would restart the loop without popping anything.

Reply:
Ahh so like if the entire try/catch were inside the while loop, makes sense.  All good 🙂